bouncer/tests/polkadot_runtime_update.ts

```diff
@@ -0,0 +1,263 @@
+      'Preimage hash mismatch',
+    );
+    console.log(`Preimage submitted: ${preimageHash}`);
+  }
+
+  // Submit the proposal
+  const observeDemocracyStarted = observeEvent('democracy:Started', polkadot);
+  const amount = amountToFineAmount(PROPOSAL_AMOUNT, assetDecimals.DOT);
+  console.log(`Submitting proposal with amount: ${PROPOSAL_AMOUNT}`);
+  const democracyStartedEvent = await submitAndGetEvent(
+    polkadot.tx.democracy.propose({ Legacy: preimageHash }, amount),
+    polkadot.events.democracy.Proposed,
+  );
+  const proposalIndex = democracyStartedEvent.data[0];
+  console.log(`proposal submitted: index ${proposalIndex}`);
+
+  // Wait for the democracy started event
+  console.log('Waiting for voting to start...');
+  await observeDemocracyStarted; // FIXME: Sometimes this event is not getting hit?
+
+  // Vote for the proposal
+  const observeDemocracyPassed = observeEvent('democracy:Passed', polkadot);
+  const observeDemocracyNotPassed = observeEvent('democracy:NotPassed', polkadot);
+  const observeSchedulerDispatched = observeEvent('scheduler:Dispatched', polkadot);
+  const vote = { Standard: { vote: true, balance: amount } };
+  await submitAndGetEvent(
+    polkadot.tx.democracy.vote(proposalIndex, vote),
+    polkadot.events.democracy.Voted,
+  );
+  console.log(`voted for proposal ${proposalIndex}`);
+
+  // Wait for it to pass
+  await Promise.race([observeDemocracyPassed, observeDemocracyNotPassed])
+    .then((event) => {
+      if (event.name.method !== 'Passed') {
+        throw new Error(`Democracy failed for runtime upgrade. ${proposalIndex}`);
+      }
+    })
+    .catch((error) => {
+      console.error(error);
+      process.exit(-1);
+    });
+  console.log('Democracy manifest! waiting for a succulent scheduled runtime upgrade...');
+
+  // Wait for the runtime upgrade to complete
+  const schedulerDispatchedEvent = await observeSchedulerDispatched;
+  if (schedulerDispatchedEvent.data.result.Err) {
+    console.log('Runtime upgrade failed');
+    handleDispatchError({
+      dispatchError: JSON.stringify({ module: schedulerDispatchedEvent.data.result.Err.Module }),
+    });
+    process.exit(-1);
+  }
+  console.log('-- Polkadot runtime upgrade complete --');
+}
+
+async function randomPolkadotSwap(): Promise<void> {
+  const assets: Asset[] = [Assets.BTC, Assets.ETH, Assets.USDC, Assets.FLIP];
+  const randomAsset = assets[Math.floor(Math.random() * assets.length)];
+
+  let sourceAsset: Asset;
+  let destAsset: Asset;
+
+  if (Math.random() < 0.5) {
+    sourceAsset = Assets.DOT;
+    destAsset = randomAsset;
+  } else {
+    sourceAsset = randomAsset;
+    destAsset = Assets.DOT;
+  }
+
+  await testSwap(sourceAsset, destAsset, undefined, undefined, undefined, undefined, false);
+  swapsComplete++;
+}
+
+async function doPolkadotSwaps(): Promise<void> {
+  console.log('Running polkadot, new swap every 1 second');
+  while (!runtimeUpgradeComplete) {
+    randomPolkadotSwap();
```

Comment:
Got this error:
```
Preimage hash: 0x7d5245955a66c95156942a2a8f4c7a3067942021c9970c9796aa74229ea542c1
2023-11-29 09:22:12        RPC-CORE: submitAndWatchExtrinsic(extrinsic: Extrinsic): ExtrinsicStatus:: 1014: Priority is too low: (12396 vs 2): The transaction has too low priority to replace another transaction already in the pool.
2023-11-29 09:22:13        RPC-CORE: submitAndWatchExtrinsic(extrinsic: Extrinsic): ExtrinsicStatus:: 1014: Priority is too low: (12396 vs 2): The transaction has too low priority to replace another transaction already in the pool.
```
and I'm guessing it's from here - haven't looked super deeply but this will make two swaps in succession within the space of a block, and I don't think it checks the mempool before submitting.

Reply:
passed this another time I tried though - which would also check out - if the timing lines up it can work.